goal_src/jak3/engine/gfx/sprite/particles/sparticle-launcher.gc

```diff
@@ -33,14 +33,2734 @@
+                 )
+             (if (nonzero? v1-2)
+                 (particle-adgif-callback (-> arg1 adgif) (the-as texture-id v1-2))
+                 )
+             )
+           )
+          (else
+            (let ((v1-4 (-> v1-0 (+ (mod (max 0 (+ (/ t0-1 a2-1) (-> arg1 user1-int16))) a0-6) 3))))
+              (if (nonzero? v1-4)
+                  (particle-adgif-callback (-> arg1 adgif) (the-as texture-id v1-4))
+                  )
+              )
+            )
+          )
+        )
+      )
+    )
+  (none)
+  )
+
+;; WARN: Return type mismatch symbol vs none.
+;; ERROR: Bad vector register dependency: vf1
+;; ERROR: Bad vector register dependency: vf2
+(defun sparticle-texture-day-night ((arg0 sparticle-system) (arg1 sparticle-cpuinfo) (arg2 sprite-vec-data-2d))
+  (local-vars
+    (v1-9 uint128)
+    (v1-10 uint128)
+    (v1-13 uint128)
+    (v1-14 uint128)
+    (v1-23 uint128)
+    (v1-24 uint128)
+    (v1-27 uint128)
+    (v1-28 uint128)
+    (v1-33 float)
+    (a0-5 float)
+    (a0-11 float)
+    (t7-0 float)
+    (t7-3 float)
+    (s3-0 float)
+    (s4-0 float)
+    )
+  (rlet ((vf1 :class vf)
+         (vf2 :class vf)
+         )
+    (let ((s2-0 (the-as (array int32) (-> arg1 user-float))))
+      (when (nonzero? s2-0)
+        (let* ((v1-1 *time-of-day*)
+               (s1-0 (if v1-1
+                         (-> v1-1 0 hours)
+                         0
+                         )
+                     )
+               (f0-0 (rand-vu))
+               )
+          (.mov s4-0 vf1)
+          (.mov s3-0 vf2)
+          (cond
+            ((or (< s1-0 6) (< 18 s1-0))
+             (let ((a1-1 (-> s2-0 7)))
+               (when (nonzero? a1-1)
+                 (let ((v1-6 f0-0))
+                   (.mov vf2 v1-6)
+                   )
+                 (let ((v1-8 (the-as uint128 (make-u128 0 (-> s2-0 9)))))
+                   (.pextlb v1-9 0 v1-8)
+                   )
+                 (.pextlb v1-10 0 v1-9)
+                 (.mov vf1 v1-10)
+                 (.itof.vf vf1 vf1)
+                 (.mul.x.vf vf1 vf1 vf2)
+                 (let ((v1-12 (the-as uint128 (make-u128 0 (-> s2-0 8)))))
+                   (.pextlb v1-13 0 v1-12)
+                   )
+                 (.pextlb v1-14 0 v1-13)
+                 (.mov vf2 v1-14)
+                 (.itof.vf vf2 vf2)
+                 (.add.vf vf1 vf1 vf2)
+                 (let ((v1-15 (-> arg1 flags-s32)))
+                   (when (nonzero? (-> s2-0 10))
+                     (.lvf vf2 (&-> *time-of-day-context* current-prt-color quad))
+                     (.mul.vf vf1 vf1 vf2)
+                     (.mov a0-5 vf1)
+                     )
+                   (let ((v1-16 (logand v1-15 #x4000)))
+                     (.mov t7-0 vf1)
+                     (let ((v1-17 (sar v1-16 14)))
+                       ;; og:preserve-this
+                       (set! (-> arg2 r-g-b-a quad) (the-as uint128 (logior (logand (the-as uint t7-0) (the-as uint -2)) v1-17)))
```

Comment:
actually - i think this relies on 64-bit ands leaving the upper 64 bits alone... maybe look at how we did this in jak 2? or fine to leave for now.